spec/fixtures/generate_external_certs.sh

```diff
@@ -0,0 +1,337 @@
+    -infiles $CSR_DIR/site_$1.csr
+}
+
+function create_site_crt_selfsigned() { # $1 domain, $2 message_digest, $3 rsa_bits, $4 good/expired, $5 subject, $6 altnames, $7 ipaltnames
+  echo "Generating site key \"$1\""
+  openssl genrsa -out $DIR_KEYS/site_$1.key $3 2>/dev/null
+  
+  init_root_ca dummy "/CN=US/O=US/OU=US Unit"
+
+  FILE=/tmp/ca_gen/site_$1.req.conf
+  site_request_conf $FILE $3
+  if [ "$6" != "0" ]; then append_altnames    $6 $FILE; fi
+  if [ "$7" != "0" ]; then append_ip_altnames $7 $FILE; fi
+  
+  openssl rsa -in $DIR_KEYS/site_$1.key -out $DIR_KEYS/site_rsa_$1.key
+  openssl req -new -sha256 -key $DIR_KEYS/site_$1.key -subj "$5" -out $CSR_DIR/site_$1.csr -config $FILE
+
+  cp $DIR_CONF/ca_dummy.conf $DIR_CONF/site_$1.conf
+  
+  # Generate an extensions file; not extending default config since x509 command doesn't have a way to specify a
+  # full config file that I see.
+  #echo -e "[req_ext]\nsubjectAltName = @alt_names\n" >> conf/site_$1.conf
+  #echo -e "[alt_names]" >> conf/site_$1.conf
+  if [ "$6" != "0" ]; then append_altnames    $6 $DIR_CONF/site_$1.conf; fi
+  if [ "$7" != "0" ]; then append_ip_altnames $7 $DIR_CONF/site_$1.conf; fi
+  
+  gen_start_end $4
+  
+  #openssl x509 -req \
+  #  -in $CSR_DIR/site_$1.csr \
+  #  -$2 \
+  #  -extensions req_ext \
+  #  -out certs/site_$1.crt \
+  #  -signkey keys/site_rsa_$1.key \
+  #  -extfile conf/site_$1.conf
+  
+  openssl ca -batch \
+    -config $DIR_CONF/site_$1.conf \
+    -out $DIR_CERTS/site_$1.crt \
+    -extensions req_ext \
+    -selfsign \
+    -keyfile $DIR_KEYS/site_rsa_$1.key \
+    -md $2 \
+    -startdate $START_DATE -enddate $END_DATE \
+    -infiles $CSR_DIR/site_$1.csr
+}
+
+# Note that an altname is specified below. If req_ext is used, then this file cannot be used as is, because an empty
+# alt_names section is not allowed. It is done this way so the file can be copied and a list of altnames
+# be appended to the end of the copy.
+function generate_conf() { # $1 ca_name
+  cat >$DIR_CONF/ca_$1.conf << EOL
+[ca]
+default_ca = default
+[default]
+certs = $DIR_CADATA/$1/
+new_certs_dir = $DIR_CADATA/$1/db.certs
+database = $DIR_CADATA/$1/db.index
+serial = $DIR_CADATA/$1/db.serial
+certificate = $DIR_CERTS/ca_$1.crt
+private_key = $DIR_KEYS/ca_$1.key
+default_days = 365
+default_crl_days = 30
+default_md = sha256
+preserve = no
+RANDFILE = $DIR_CADATA/$1/db.random
+policy = default_policy
+[default_policy]
+countryName = optional
+stateOrProvinceName = optional
+localityName = optional
+organizationName = supplied
+organizationalUnitName = supplied
+commonName = supplied
+emailAddress = optional
+[ v3_ca ]
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid:always,issuer
+basicConstraints = critical,CA:true
+[req_ext]
+subjectAltName = @alt_names
+[alt_names]
+EOL
+}
+
+read -r -d '' ALTNAMES_VELUM_LN << EOM
+admin.devenv.caasp.suse.net
+admin
+admin.infra.caasp.local
+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
```

Comment:
It might be a good idea to use a more recognizable string to match as it is hard to match and see that the right number of a's are in use. Something like aaaabbbbccccddddeeee perhaps? Maybe even just using a short string. The long name is used because when the system generates a name it is a long randomly generated string of this length.

Additionally this name is the long uuid of the master node. So perhaps "master_long_uuid_24_chars" would be another option ( assuming it is a uuid; I haven't looked carefully at what generates the random strings to see it is in fact a uuid; it may or may not be )

Reply:
Leaving this alone for now as I don't think it is worth changing. It works fine as is and isn't all that confusing.

Reply:
The node name doesn't have to be included in any of these certs, as far as I can tell, so this is optional anyway.